### ISSUE TITLE
The "parse" function has inconsistent error handling

### ISSUE THREAD

**Issue by maerics** [kelektiv/node-uuid#110]
The `parse(...)` function appears to return a seemingly valid result when given any string, regardless of whether it contains a valid UUID or not:

``` javascript
var nodeUuid = require('node-uuid'); // v1.4.3
nodeUuid.parse(''); // => [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
nodeUuid.parse('this is not a uuid'); // => [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
nodeUuid.parse('ffff'); // => [255, 255, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
nodeUuid.parse(new Object); // => TypeError: Object #<Object> has no method 'toLowerCase'
nodeUuid.parse(nodeUuid.v4()); // => [145, 125, 225, 14, ... ]
```

How would a user of this library know whether or not the argument is a valid UUID or not?

Would it make sense to return `null` or throw an exception indicating that the argument is not a parseable UUID string?

**Comment by tracker1** [comment]
+1 for `null` or maybe `NaN` (Invalid Date wraps NaN).

**Comment by coolaj86** [comment]
What is the expected behavior? Should it only parse exactly matching strings (including hyphens in the correct places) or otherwise throw an error?

**Comment by coolaj86** [comment]
hmmm it's being tested to behave this way: https://github.com/broofa/node-uuid/blob/master/test/test.js#L129

Any change to this would definitely require a bump to v2.0.0, but I agree that the suggested behavior sounds better than the current behavior.

Thoughts @broofa?

**Comment by broofa** [comment]
That parse/unparse methods are publicly exposed is a bit of a mistake, iirc. They were methods I found useful in the original version of this lib and I figured others might find them useful as well, so added them to the public API, but didn't give a lot of thought to it.  They were certainly never intended to be a rigorous method of validating uuids.

This is one of the things I'd like to see addressed in the next major version.  In fact if you look at the parseFields method in the v2.0.0 branch you'll find a more rigorous implementation that's more in keeping I with what I think the public API for parsing/validation should be.
